src/Symfony/Component/Translation/Formatter/MessageFormatterInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Formatter;
+
+/**
+ * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+interface MessageFormatterInterface
+{
+    /**
+     * Formats a localized message pattern with given arguments.
+     *
+     * @param string $message    The message (may also be an object that can be cast to string)
+     * @param string $locale     The message locale
+     * @param array  $parameters An array of parameters for the message
+     *
+     * @return string
+     */
+    public function format($message, $locale, array $parameters = array());
```

Comment:
see the use-case [IntlMessageFormatter](https://github.com/aitboudad/symfony/blob/7c3ac84d41570270b8cfcaae4c87a0b5c1384c86/src/Symfony/Component/Translation/Formatter/IntlMessageFormatter.php#L25) (https://github.com/symfony/symfony/pull/20007)